gobblin-data-management/src/main/java/gobblin/data/management/retention/policy/ProxyableDatasetVersionPredicate.java

```diff
@@ -0,0 +1,40 @@
+package gobblin.data.management.retention.policy;
+
+import java.util.Properties;
+
+import com.google.common.base.Predicate;
+
+import lombok.AllArgsConstructor;
+import lombok.NonNull;
+
+import org.apache.hadoop.fs.Path;
+
+import gobblin.configuration.ConfigurationKeys;
+import gobblin.data.management.retention.version.DatasetVersion;
+import gobblin.data.management.retention.version.StringDatasetVersion;
+import gobblin.util.ProxiedFileSystemUtils;
+
+
+@AllArgsConstructor
+public class ProxyableDatasetVersionPredicate implements Predicate<DatasetVersion> {
```

Comment:
This class is hacky. `ProxiedFileSystemUtils.canProxyAs()` really should not be the way of figuring out whether a directory has the name of a user, nor does the name of the class imply that that's what it is doing.
Can we move either clean it up, or move the class to an internal repo?